### PR TITLE
Add a file to keep track of deprecated features

### PR DESCRIPTION
**Pull request by victorlin**
## Description of proposed changes

Descriptive text inspired by an analogous file used in the Commander.js repo.¹

¹ <https://github.com/tj/commander.js/blob/f1ae2db8e2da01d6efcbfd59cbf82202f864b0c1/docs/deprecated.md>

## Related issue(s)

- #1266
- Used by 4c0900542349e24a2a82518a7db6d9a245a5e070

## Checklist

- [x] Checks pass
- [x] If making user-facing changes, add a message in [CHANGES.md](https://github.com/nextstrain/augur/blob/HEAD/CHANGES.md) summarizing the changes in this PR

### CODE REVIEW

**Review by jameshadfield — APPROVED** [review]

**Comment by victorlin** [comment]
I just realized we have the [Breaking Changes](https://docs.nextstrain.org/projects/augur/page/releases/breaking.html) docs page, which overlaps with DEPRECATED.md. I'll merge the two together.

File: docs/releases/breaking.md
Comment (victorlin) [inline docs/releases/breaking.md:1]: @jameshadfield I've added 02f76185ea29cbfa90688a36f68ebe313353c88a and rebased - can you re-review?

Comment (jameshadfield) [reply docs/releases/breaking.md:1]: In principle the `breaking.md` is a great resource. In practice nothing's been added to it for four years! The combination of helpful deprecation warnings when running the code + a changelog detailing what's been removed (and doing so in an easily-searchable fashion) is a better path to actually documenting this.

**Review by jameshadfield — APPROVED** [review]

File: docs/releases/breaking.md
Comment (victorlin) [reply docs/releases/breaking.md:1]: Good point on these things being useful in principle but not purpose. Pushed 171986e5d336de1b5f2462bae937dceffa23de94 to hopefully prevent that from happening with this new file.

File: DEPRECATED.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+`augur export v2` was introduced in Augur version 6.0.0. Migrate by following
+the [official guide](https://docs.nextstrain.org/projects/augur/page/releases/migrating-v5-v6.html).
+
+## `augur ancestral --output`
+
+*Deprecated in version 5.2.0 (December 2019). Removed in version 7.0.0 (April
+2020).*
+
+`--output` was been replaced by `--output-node-data` to accommodate the addition
+of `--output-sequences`.
```
Comment (victorlin) [inline DEPRECATED.md:21]: Oops, typo - fixed in e7422911a481147689ab54d46fc2300fcfef0fb8.

**Pull request merged**